observer.py

```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+from random import choice
+
+class IObservable:
+    @abstractmethod
+    def add(self, observer):
+        pass
+
+    @abstractmethod
+    def remove(self, observer):
+        pass
+
+    @abstractmethod
+    def notify(self, topic):
+        pass
+
+class IObserver:
+    @abstractmethod
+    def update(self, topic):
+        pass
+
+class TopicStore:
+    def __init__(self, topics):
+        self.topics = topics
+
+    def is_topic_available(self, topic):
+        return topic in self.topics
+
+    def remove_topic(self, topic):
+        self.topics.remove(topic)
+
+class ConcreteTeacher(IObservable):
+    def __init__(self, topic_store):
+        self.topic_store = topic_store
+        self.observers = []
+
+    def add(self, observer):
+        self.observers.append(observer)
+
+    def remove(self, observer):
+        self.observers.remove(observer)
+
+    def notify(self, topic):
+        for observer in self.observers:
+            observer.update(topic)
+
+    def remove_topic(self, topic):
+        if self.topic_store.is_topic_available(topic):
+            self.topic_store.remove_topic(topic)
+            self.notify(topic)
+
+class ConcreteStudent(IObserver):
+    def __init__(self, name):
+        self.name = name
+        self.selected_topic = None
+
+    def select_random_topic(self, teacher):
```

Comment:
This teacher param is observable and the observer class is supposed to know its observable beforehand. 
```python
observer.register(observable)
```
This line should store the object to each other.